### PR TITLE
Break out Repo access from API.ReleaseController

### PR DESCRIPTION
**Pull request by wojtekmach**
See #386

### CODE REVIEW

File: web/models/releases.ex [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    # Delete release tarball
+    HexWeb.Store.delete(nil, :s3_bucket, key, [])
+
+    # Delete relevant documentation (if it exists)
+    if release.has_docs do
+      HexWeb.Store.delete(nil, :s3_bucket, "docs/#{name}-#{version}.tar.gz", [])
+      paths = HexWeb.Store.list(nil, :docs_bucket, Path.join(name, version))
+      HexWeb.Store.delete_many(nil, :docs_bucket, Enum.to_list(paths), [])
+      HexWeb.API.DocsController.publish_sitemap
```
Comment (wojtekmach) [inline web/models/releases.ex]: I think we shouldn't call controllers from these modules (only the other way around) and so I was thinking about introducing `Sitemaps` modules that would contain this publishing logic. If that sounds good I'd like to do that in a separate PR to keep this one focused on just moving stuff around.

Comment (ericmj) [inline web/models/releases.ex]: Yes, agreed.

File: web/models/releases.ex [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  end
+
+  defp publish_release(multi, body) do
+    Ecto.Multi.run(multi, :assets, fn %{package: package, release: release} ->
+      push_assets(package, release.version, body)
+      {:ok, :ok}
+    end)
+  end
+
+  defp push_assets(package, version, body) do
```
Comment (ericmj) [inline web/models/releases.ex]: Not needed in this PR but maybe asset handling should be in its own module. Not sure though, up to you, you would have better feel for it since you are doing the refactorings :).

Comment (wojtekmach) [inline web/models/releases.ex]: yeah, I wanted to propose something after merging this PR (and an upcoming one for DocsController). I haven't looked at the assets code across the app to closely yet, but perhaps `HexWeb.Assets` (that under the hood uses CDN and Store) would be good.

**Comment by ericmj** [comment]
LGTM. Ready to merge?

**Comment by wojtekmach** [comment]
@ericmj from my side it's ready to go out. For the things we discussed here I'll open up separate PRs.
